### PR TITLE
Direction arrow

### PR DESCRIPTION
**Pull request by ErikSin**
### Added Directional Arrow:

User is able to turn on and off directional arrow in settings.

Kapture 2022-01-23 at 20 44 57

### CODE REVIEW

File: src/frontend/sharedComponents/Map/MapView.js [hunk 1]
```diff
@@ -391,7 +393,7 @@ class MapView extends React.Component<Props, State> {
             )}
             {styleType === "fallback" ? <OfflineMapLayers /> : null}
             {locationServicesEnabled ? (
-              <MapboxGL.UserLocation
+              <UserLocation
                 visible={isFocused}
                 minDisplacement={MIN_DISPLACEMENT}
```
Comment (ErikSin) [inline src/frontend/sharedComponents/Map/MapView.js:397]: I  converted this to a functional component so I could use the `useSettingsValue` hook

File: src/frontend/sharedComponents/Map/MapView.js [hunk 2]
```diff
@@ -6,17 +6,19 @@ import ScaleBar from "react-native-scale-bar";
...
+import IconButton from "../IconButton";
+import type { LocationContextType } from "../../context/LocationContext";
+import type { ObservationsMap } from "../../context/ObservationsContext";
+import type { MapStyleType } from "../../hooks/useMapStyle";
+import { useIsFullyFocused } from "../../hooks/useIsFullyFocused";
+import bugsnag from "../../lib/logger";
+import config from "../../../config.json";
+import Loading from "../Loading";
+import OfflineMapLayers from "../OfflineMapLayers";
+import SettingsContext from "../../context/SettingsContext";
```
Comment (achou11) [inline src/frontend/sharedComponents/Map/MapView.js]: Remove unused import

File: src/frontend/hooks/useSettingsValue.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import * as React from "react";
+import SettingsContext, { SettingsState } from "../context/SettingsContext";
+
+export default function useSettingsValue(key: keyof SettingsState) {
+  const [state] = React.useContext(SettingsContext);
+  return state[key];
+}
```
Comment (achou11) [inline src/frontend/hooks/useSettingsValue.ts]: To improve type inference, we can add a generic here:

```suggestion
export default function useSettingsValue<K extends keyof SettingsState>(
  key: K
): SettingsState[K] {
  const [state] = React.useContext(SettingsContext);
  return state[key];
}
```

File: src/frontend/sharedComponents/Map/UserLocation.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+interface UserLocationProps {
+  visible: boolean;
+  minDisplacement: number;
+}
+
+export const UserLocation = ({
+  visible,
+  minDisplacement,
+}: UserLocationProps) => {
+  const directionalArrow = useSettingsValue("directionalArrow") as boolean;
```
Comment (achou11) [inline src/frontend/sharedComponents/Map/UserLocation.tsx]: Can avoid the type assertion here by updating `useSettingsValue` here: https://github.com/digidem/mapeo-mobile/pull/914/files#r790968391

File: src/frontend/screens/Settings/MapSettings.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
+import React, { useContext } from "react";
+import { FormattedMessage, defineMessages } from "react-intl";
+import { View, Text, Switch } from "react-native";
```
Comment (achou11) [inline src/frontend/screens/Settings/MapSettings.tsx]: Can remove unused `View` and `Text` imports here

File: src/frontend/screens/Settings/MapSettings.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import { useNavigation } from "react-navigation-hooks";
+import { NavigationStackScreenComponent } from "react-navigation-stack";
+import SettingsContext from "../../context/SettingsContext";
+import {
+  DARK_BLUE,
+  LIGHT_BLUE,
+  VERY_LIGHT_BLUE,
```
Comment (achou11) [inline src/frontend/screens/Settings/MapSettings.tsx]: Can remove unused `LIGHT_BLUE` and `VERY_LIGHT_BLUE` imports here

File: src/frontend/screens/Settings/Settings.js [hunk 7]
```diff
@@ -126,6 +131,17 @@ const Settings = () => {
...
         ></ListItemText>
       </ListItem>
+      <ListItem
+        onPress={() => navigate("MapSettings")}
+        testID="settingsMapSettingsButton"
+      >
+        <ListItemIcon iconName="map" />
+        <ListItemText
+          primary={<FormattedMessage {...m.mapSettings} />}
+          secondary="---------"
```
Comment (achou11) [inline src/frontend/screens/Settings/Settings.js:141]: Can we either omit or add something more meaningful here?

File: src/frontend/screens/Settings/MapSettings.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+  ),
+  headerLeft: ({ onPress }) =>
+    onPress && (
+      <IconButton onPress={onPress}>
+        <BackIcon />
+      </IconButton>
+    ),
+});
+
+const useArrowListItem = () => {};
```
Comment (achou11) [inline src/frontend/screens/Settings/MapSettings.tsx]: Remove if not needed

File: src/frontend/screens/Settings/Settings.js (hunk 7 above)
Comment (ErikSin) [reply src/frontend/screens/Settings/Settings.js:141]: This was in the mockups

File: src/frontend/screens/Settings/Settings.js [hunk 9]
```diff
@@ -98,6 +98,11 @@ const m = defineMessages({
     defaultMessage: "App Passcode and Device Security",
     description: "Description of security button in settings",
   },
+  mapSettings: {
+    id: "screens.Settings.mapSettingd",
```
Comment (achou11) [inline src/frontend/screens/Settings/Settings.js]: small typo here?
```suggestion
    id: "screens.Settings.mapSettings",
```

File: src/frontend/screens/Settings/Settings.js (hunk 7 above)
Comment (achou11) [reply src/frontend/screens/Settings/Settings.js:141]: ok i would ask to confirm that's what we want, but not a blocker for the PR

Comment (ErikSin) [reply src/frontend/screens/Settings/Settings.js:141]: Lets just merge this so  we can get the Release, and we can fix it if Afterwards if sabella decides to put something else

**Review by achou11 — APPROVED** [review]
Some relatively minor comments but otheriwse lgtm

**Pull request merged**